builtin/credential/github/backend.go

```diff
@@ -69,7 +70,14 @@ func (b *backend) Client(token string) (*github.Client, error) {
 		tc = oauth2.NewClient(ctx, &tokenSource{Value: token})
 	}
 
-	return github.NewClient(tc), nil
+	client := github.NewClient(tc)
```

Comment:
nit: I think if we used [github.NewEnterpriseClient](https://pkg.go.dev/github.com/google/go-github/v45/github#NewEnterpriseClient) and passed the baseURL and uploadURL to this method we could potentially remove a lot of the url parsing in this file, path_config.go and path_login.go. Feel free to ignore this if you don't think it is worth it.

Reply:
That's a good point, I tried it but unfortunately I don't think we can use NewEntepriseClient while keeping the current default baseURL fallback behavior.

If base_url is not provided in the plugin configuration, we fallback on the default constant defaultBaseURL from the github package which is not visible to vault.

If NewEnterpriseClient would fallback on defaultBaseURL in the event of receiving an empty or nil baseURL parameter I'd be 100% on board to remove all this parsing. As is it however, I don't think it's possible unless we copy the constant from the external go-github dependency which would not be ideal.

I hope it make sense. Let me know if I misunderstood.

Reply:
makes sense, thanks!